shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparer.java

```diff
@@ -90,27 +89,25 @@ private void prepareAndCheckTargetWithLock(final RuleAlteredJobContext jobContex
         String lockName = "prepare-" + jobConfig.getJobId();
         LockContext lockContext = PipelineContext.getContextManager().getInstanceContext().getLockContext();
         LockDefinition lockDefinition = new ExclusiveLockDefinition(lockName);
-        if (lockContext.tryLock(lockDefinition, 3000)) {
+        if (lockContext.tryLock(lockDefinition, 180000)) {
             log.info("try lock success, jobId={}, shardingItem={}", jobConfig.getJobId(), jobContext.getShardingItem());
             try {
-                prepareAndCheckTarget(jobContext);
+                JobProgress jobProgress = RuleAlteredJobAPIFactory.getInstance().getJobProgress(jobContext.getJobId(), jobContext.getShardingItem());
+                boolean prepareFlag = null == jobProgress || JobStatus.RUNNING.equals(jobProgress.getStatus()) || JobStatus.PREPARING_FAILURE.equals(jobProgress.getStatus());
+                if (prepareFlag) {
+                    log.info("execute prepare, jobId={}, shardingItem={}", jobConfig.getJobId(), jobContext.getShardingItem());
+                    jobContext.setStatus(JobStatus.PREPARING);
+                    RuleAlteredJobAPIFactory.getInstance().persistJobProgress(jobContext);
+                    prepareAndCheckTarget(jobContext);
+                    jobContext.setStatus(JobStatus.PREPARE_SUCCESS);
+                    for (int i = 0; i <= jobContext.getJobConfig().getJobShardingCount(); i++) {
+                        RuleAlteredJobAPIFactory.getInstance().persistJobProgress(jobContext);
+                    }
```

Comment:
Seems `persistJobProgress(jobContext)` in for loop will cause all sharding item's progress will overwritten by current sharding item's progress, it might cause issue.

Could we just update every sharding item progress's status? Try `void updateShardingJobStatus(String jobId, int shardingItem, JobStatus status);`